core/vm/interpreter.go

```diff
@@ -17,20 +17,63 @@
 package vm
 
 import (
+	"fmt"
 	"hash"
+	"io"
 	"sync/atomic"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/math"
 	"github.com/ethereum/go-ethereum/log"
 )
 
+type InstrumenterLog struct {
+	Pc          uint64 `json:"pc"`
+	Op          OpCode `json:"op"`
+	TimeNs      int64  `json:"timeNs"`
+	TimerTimeNs int64  `json:"timerTimeNs"`
+}
+
+type InstrumenterLogger struct {
+	Logs            []InstrumenterLog
+	LastCaptureTime int64
+
+	// worker fields, just to avoid reallocation of local vars
+	TimeSincePrevious int64
+	TimerRuntime      int64
```

Comment:
Maybe consider renaming `TimerRuntime`, as it may be a bit misleading. Runtime is usually considered  as _runtime environment_ of some program. My proposition is `TimerDrift` or `TimerErr` or even `TimerMeasurementDuration`.

Reply:
I'll try to align with the names from ruleset :+1: